### ISSUE TITLE
Add pretty print to SqlResult

### ISSUE THREAD

**Issue by PavloBida** [hazelcast/hazelcast#19386]
It would be a nice to have feature to have some kind of print() method in [SqlResult](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/sql/SqlResult.java) class to have the result of executed query pretty printed in the console. Exactly like we have it in Hazelcast CLI:
```
+--------------------+--------------------+
|               __key|c0                  |
+--------------------+--------------------+
|                   2|test2               |
|                   0|test0               |
|                   1|test1               |
+--------------------+--------------------+
3 row(s) selected
```
Note! That pretty print shouldn't call the iterator() or at least preserve such a possibility for the user.

**Comment by viliam-durina** [comment]
>Note! That pretty print shouldn't call the iterator() or at least preserve such a possibility for the user.

It must call the `iterator()` method and the result is unusable after this. It also won't work if `iterator()` was already called. The reason is that you can iterate the result only once. We can document this.
